src/Farmer/Arm/Dashboard.fs

```diff
@@ -0,0 +1,163 @@
+    isAdapter = None
+    defaultMenuItemId = Some "overview"
+}
+
+/// Generates a virtualMachinePart
+let generateWebtestResultPart (applicationInsightsName:string) = {
+    ``type`` = "Extension/AppInsightsExtension/PartType/AllWebTestsResponseTimeFullGalleryAdapterPart"
+    inputs = [ {| name = "ComponentId"; value = {| Name = applicationInsightsName; SubscriptionId = "[ subscription().subscriptionId ]"; ResourceGroup = "[ resourceGroup().id ]" |} |} ]
+    settings = None
+    filters = None
+    asset = Some {| idInputName = "ComponentId"; ``type`` = "ApplicationInsights" |}
+    isAdapter = Some true
+    defaultMenuItemId = None
+}
+
+/// Generates a MetricsChartPart for a resource given in parameters
+let generateMetricsChartPart (chartProperties:{| resourceId:ResourceId; metrics: ChartResources list; interval : ChartDurationInterval |}) = {
+    ``type`` = "Extension/Microsoft_Azure_Monitoring/PartType/MetricsChartPart"
+    inputs = [ {| name = "queryInputs"
+                  value = {| id = chartProperties.resourceId.ArmExpression.Eval(); chartType = 0;
+                             timespan = {| duration = match chartProperties.interval with | ISO8601DurationFormat dur -> dur
+                                           start = null; ``end`` = null |}
+                             metrics = chartProperties.metrics |> List.map(function 
+                                            ChartResouce m -> {| name = m; resourceId = chartProperties.resourceId.ArmExpression.Eval() |})
+                          |} |}  ]
+    settings = None
+    filters = None
+    asset = None
+    isAdapter = None
+    defaultMenuItemId = None
+}
+
+/// Generates a MonitorChartPart
+let generateMonitorChartPart (chartProperties : {| chartInputs:obj list; chartSettings: obj option; filters : obj option |}) = {
+    ``type`` = "Extension/HubsExtension/PartType/MonitorChartPart"
+    inputs = [ box <| {| name = "sharedTimeRange"; isOptional = true |};
+               box <| {| name = "options"
+                         value = {| v2charts = true
+                                    charts = [ chartProperties.chartInputs ] |} |} ]
+    settings = Some ({| content = {| options = {| chart = chartProperties.chartSettings |} |} |})
+    filters = chartProperties.filters
+    asset = None
+    isAdapter = None
+    defaultMenuItemId = None
+}
+
+type Dashboard =
+    { Name : ResourceName
+      Title : string option
+      Location : Location
+      Metadata : DashboardMetadata
+      LensParts : LensPart list
+      Dependencies : Set<ResourceId> }
+    interface IArmResource with
+        member this.ResourceId = dashboard.resourceId this.Name
+        member this.JsonModel =
+
+            let lensesPartsObjJson =
+                // Parts-list is not array, instead it's "parts": { "0": { }, "1": { }, ... }
+                let lensesJsons = this.LensParts |> List.mapi(fun idx item -> $"\"{idx}\": {item |> Farmer.Serialization.toJson }")
```

Comment:
We have parts in F# list. 
But sadly the JSON is not a list of `"parts": [ .. ]` but `"parts" : { ... }".
So the question is: How do you convert F# list to an object where the object has dynamic number of properties?
So basically:
```fsharp
type Parts() =
    member __.``0`` = ...
    member __.``1`` = ...
    member __.``2`` = ...
   ...
```
F# is a strongly typed language. So that is painful anyway, instead of writing custom (reflection or expression or ...)-code, I just used the current JSON serialization to do that:
LensParts translates F# list to JSON dynamic object, which is using Farmer serialization to convert that back to F#. (And then later Farmer will convert that back to JSON.)

Reply:
The new API got rid of this JSON structure and used array, so I could remove this piece of code totally.